File: trunk/Templates/Examples/NHibernate/VisualBasic/NHibVbSample.Data/Base/UnitTestsBase.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports NUnit.Framework
Imports NHibVbSample.Generated.ManagerObjects
Imports NHibVbSample.Generated.BusinessObjects

Namespace NHibVbSample.Generated.Base
	Public Class UNuitTestBase
		Protected managerFactory As IManagerFactory = New ManagerFactory()

		<SetUp()> _
		Public Sub SetUp()
			NHibernateSessionManager.Instance.Session.BeginTransaction()
		End Sub
		<TearDown()> _
		Public Sub TearDown()
			NHibernateSessionManager.Instance.Session.RollbackTransaction()
		End Sub
	End Class
End Namespace

File: trunk/Projects/Framework-Samples/NHibernate/CSharp/NHibCsSample.Data/Base/UnitTestsBase.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using NHibCsSample.Generated.ManagerObjects;
using NHibCsSample.Generated.BusinessObjects;

namespace NHibCsSample.Generated.Base
{
    public class UNuitTestBase
    {
        protected IManagerFactory managerFactory = new ManagerFactory();

        [SetUp]
        public void SetUp()
        {
            NHibernateSessionManager.Instance.Session.BeginTransaction();
        }
        [TearDown]
        public void TearDown()
        {
            NHibernateSessionManager.Instance.Session.RollbackTransaction();
        }
    }
}